spec/support/features/session_helper.rb

```diff
@@ -162,7 +162,7 @@ def acknowledge_and_confirm_recovery_code
       click_on button_text, class: 'recovery-code-continue'
 
       code_words.size.times do |index|
-        fill_in "recovery-#{index}", with: code_words[index].downcase
+        fill_in "recovery-#{index}", with: code_words[index].downcase + 'xxxx'
```

Comment:
I hear you, but I think this is required for release (at least the issue is labeled as such), to adhere to the [Accept known good](https://www.owasp.org/index.php/Data_Validation#Accept_known_good) principle. See Justin's comment: https://github.com/18F/identity-private/issues/1602#issuecomment-285838170

We know that the code can only be 16 characters and can only be alphanumeric, so according to that issue, to minimize the risk of someone sending the server characters outside of the ones we should accept, we should validate the code before we check if it matches.
https://github.com/18F/identity-private/issues/1602#issuecomment-285838170